src/urllib3/response.py

```diff
@@ -742,16 +752,31 @@ def from_httplib(
         """
         headers = r.msg
 
+        try:
+            assert_header_parsing(headers)
+        except (HeaderParsingError, TypeError) as hpe:
+            log.warning(
+                "Failed to parse headers (url=%s): %s",
+                _absolute_url(**response_kw),
```

Comment:
So `_absolute_url` needs variables that are mostly found in `HTTPConnectionPool`.  It seems like `pool` is always passed in `response_kw`? But is it really always passed? 

I'm not familiar with how the internals of urlib3 work. Will `HTTPConnection` 's always belong to a connection pool? 

It didn't break any tests, so I assumed it was safe to move this function here and count on `pool` always being in `response_kw` kwargs. But I was also wondering why `host`, `scheme` and `port` were not available directly from `HTTPConnection`. Also shouldn't `HTTPConnection` have an attribute/instance variable linking it to the `HTTPConnectionPool` that created it? Something like `conn.pool`?

Just some of my thoughts while working my way through an unfamiliar codebase.

Reply:
You can reconstruct the URL using the connection's host, port, scheme, and the request target (`request_url` in this instance)

Reply:
I did some more weird workaround stuff 😊, but I will resolve this as its been moved to another file